force-app/main/default/classes/RD2_ChangeHistoryService.cls

```diff
@@ -0,0 +1,136 @@
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2021
+* @group Recurring Donations
+* @description All services specific to Recurring Donation Change History
+*
+*/
+public without sharing class RD2_ChangeHistoryService {
+
+    /***
+    * @description Determines if Change History is enabled for the org
+    */
+    public Boolean isChangeHistoryEnabled {
+        get {
+            if (isChangeHistoryEnabled == null) {
+                isChangeHistoryEnabled = new RD2_Settings().isChangeHistoryEnabled();
+            }
+            return isChangeHistoryEnabled;
+        }
+        set;
+    }
+
+    SObjectField rdAmountField = npe03__Recurring_Donation__c.npe03__Amount__c;
+    SObjectField rdInstallmentPeriodField = npe03__Recurring_Donation__c.npe03__Installment_Period__c;
+    SObjectField rdInstallmentFrequencyField = npe03__Recurring_Donation__c.InstallmentFrequency__c;
+    SObjectField rdStatusField = npe03__Recurring_Donation__c.Status__c;
+    SObjectField rdStatusReasonField = npe03__Recurring_Donation__c.ClosedReason__c;
+    SObjectField rdPaymentMethodField = npe03__Recurring_Donation__c.PaymentMethod__c;
+    SObjectField rdCampaignField = npe03__Recurring_Donation__c.npe03__Recurring_Donation_Campaign__c;
+
+    private List<RecurringDonationChangeHistory__c> changeHistoryList = 
+        new List<RecurringDonationChangeHistory__c>();
+
+    private class ChangeHistoryFieldSet {
+        ChangeHistoryFieldSet(SObjectField newField, SObjectField previousField) {
+            this.newField = newField;
+            this.previousField = previousField;
+        }
+
+        SObjectField newField;
+        SObjectField previousField;
+    }
+
+    private Map<SObjectField, ChangeHistoryFieldSet> recurringDonationToChangeHistoryFields = new Map<SObjectField, ChangeHistoryFieldSet>{
```

Comment:
Minor, but I think the longer var is hard to read and the shorter one gets across the same meaning
```suggestion
    private Map<SObjectField, ChangeHistoryFieldSet> rd2ChangeHistoryFields = new Map<SObjectField, ChangeHistoryFieldSet>{
```